ansible_base/authentication/backend.py

```diff
@@ -36,6 +38,11 @@ def authenticate(self, request, *args, **kwargs):
 
         for authenticator_id, authenticator_object in get_authentication_backends(last_modified).items():
             user = authenticator_object.authenticate(request, *args, **kwargs)
+
+            # Social Auth pipeline can return status string when update_user_claims fails (authentication maps deny access)
```

Comment:
I guess that this would happen when the user has selected this specific authenticator, but there is an auth map that blocks the authentication (despite it maybe being a successful auth with the external system).  So this will just hop over this intended authenticator and let the rest fail as well?  I wonder if it's better to throw an AuthForbidden exception or similar (from social auth), which will kill the process and show some sort of error message (TBD, waiting on UI update).  I guess in any case the end result will be the AuthException from determine_username_from_uid_social but maybe it's better to go directly to an exception here rather than allowing the rest of the (fruitless) activity to happen?

Reply:
The AuthException in social pipeline will cause 500 error, regarding doc, pipeline method doesn't expect any Exception.

Regarding this line, it follows the pattern which is there.  If the authenticate() is not successful, it continues to next in a loop

Other authenticators returns None or False (directly from `update_user_claims`), but the pipeline **can't`** return None or False otherwise the `authenticate()` returns the user. 

So there is no change in the main logic, just fixing the weird behaviour of expected social-django pipeline return values.

Reply:
maybe we can think about the `if not getattr(user, 'is_active', True) and user == SOCIAL_AUTH_PIPELINE_FAILED_STATUS` together, that I can add (but I guess you didn't mean that)

Reply:
Yeah OK, I went and looked at the pipeline sequence and I see this is the very last element, so no use in short circuiting it anyway.  The only reason to throw an exception would be to display the new UI message from PR 467, since I guess this continue will cycle through the rest of the authenticators and invisibly fail? What happens from a user/UI perspective if no authenticators can authenticate?

Reply:
It returns None, so authentication failed